monkey/infection_monkey/island_api_client/http_requests_facade.py

```diff
@@ -0,0 +1,99 @@
+# Retries improve reliability and slightly mitigates performance issues
+import functools
+import json
+import logging
+from enum import Enum, auto
+
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3 import Retry
+
+from common.common_consts.timeouts import MEDIUM_REQUEST_TIMEOUT
+from infection_monkey.island_api_client import (
+    IslandAPIConnectionError,
+    IslandAPIError,
+    IslandAPIRequestError,
+    IslandAPIRequestFailedError,
+    IslandAPITimeoutError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+RETRIES = 5
+
+
+class RequestMethod(Enum):
+    GET = auto()
+    POST = auto()
+    PUT = auto()
+
+
+class HTTPRequestsFacade:
```

Comment:
This facade is not intended to be used outside of this package. Fixed imports though

Reply:
What are the rules for import statements? Let's document it in the guidelines.

Reply:
I don't think there's anything specific for our project. The purpose of adding imports into `__init__.py` is to simplify the usage of a package (in this case `island_api_client`). Since `HTTPRequestsFacade` should not be used from outside this package there's no point in simplifying its import.